ts/util/table-builder.ts

```diff
@@ -60,6 +76,71 @@ export class TableBuilder<ParentType, ChildType> {
     this.reload();
   }
 
+  private addSortControls() {
+    const sortContainer = document.createElement("div");
+    sortContainer.classList.add("sortContainer");
+    this.container.appendChild(sortContainer);
+
+    const icon = makeIcon("sort");
+    icon.title = "Sort";
+    sortContainer.appendChild(icon);
+
+    const dropdown = document.createElement("select");
+    this.definition.columns
+      .filter((column) => column.sortType)
+      .forEach((column) => {
+        this.addSortOption(column, false, dropdown);
+        this.addSortOption(column, true, dropdown);
+      });
+    dropdown.addEventListener("input", (event) => {
+      const dropdown = <HTMLSelectElement>event.target;
+      const selected = dropdown.options[dropdown.selectedIndex];
+      this.sortColumn = <string>selected.getAttribute("data-column");
+      this.sortDescending = selected.getAttribute("data-order") == "desc";
+      this.reload();
+    });
+    sortContainer.appendChild(dropdown);
+  }
+
+  private addSortOption(
+    column: ColumnDefinition<ParentType, ChildType>,
+    descending: boolean,
+    dropdown: HTMLSelectElement
+  ) {
+    const option = document.createElement("option");
+    option.value = column.title + (descending ? "/desc" : "/asc");
+    option.setAttribute("data-column", column.title);
+    option.setAttribute("data-order", descending ? "desc" : "asc");
+    option.text =
+      column.title +
+      " - " +
```

Comment:
change to a semicolon? Seems to be the convention in other sorting menus

Reply:
I've seen colons, semicolons, dashes, no separator... There doesn't seem to be any established convention. I tried both ways and think that the dash looks better as it adds some extra space between the column name and ordering